lib/langchain/tool/base.rb

```diff
@@ -72,6 +72,22 @@ def tool_description
       self.class.const_get(:DESCRIPTION)
     end
 
+    #
+    # Returns the SCHEMA constant of the tool
+    #
+    # @return [Hash] tool description
+    #
+    def tool_schema
+      {
+        name: tool_name,
+        description: tool_description,
+        parameters: {
+          type: "object",
+          properties: self.class.const_get(:SCHEMA_PARAMETERS)
+        }
+      }
```

Comment:
wdyt about implementing requiered parameters? they are defined inside parameters object, and is an array of strings. possible implementation:

```
{
        name: tool_name,
        description: tool_description,
        parameters: {
          type: "object",
          properties: self.class.const_get(:SCHEMA_PARAMETERS),
          required: self.class.const_get(:REQUIRED_PARAMETERS)
        },
      }
```
and then each tool defines its own `REQUIRED_PARAMETERS` const